File: app-java/src/test/java/tech/thdev/java_udemy_sample/_01_constructor/_02_MultiConstructor.java
package tech.thdev.java_udemy_sample._01_constructor;

/**
 * Created by tae-hwan on 09/10/2016.
 */

public class _02_MultiConstructor {

    public _02_MultiConstructor(String name) {

    }

    public _02_MultiConstructor(String name, int age) {
        this(name);
    }
}

File: app-kotlin/src/test/java/tech/thdev/kotlin_udemy_sample/_01_constructor/_02_MultiConstructor.kt
package tech.thdev.kotlin_udemy_sample._01_constructor

/**
 * Created by tae-hwan on 09/10/2016.
 */

/**
 * 다중 생성자
 */

// 첫 번째 기본 생성자
class _01_Constructor(val name: String) {

    // 두 번째 생성자
    constructor(name: String, age: Int) : this(name)
}